spec/std/array_spec.cr

```diff
@@ -2041,4 +2041,18 @@ describe "Array" do
       ary.skip(-1)
     end
   end
+
+  it "#resize" do
+    ary = [1, 2, 3]
+    ary.resize(3).should eq([1, 2, 3])
+    ary.resize(4).should eq([1, 2, 3])
+    ary.remaining_capacity.should eq(4)
+    ary.shift
+    ary.resize(3).should eq([2, 3])
+    ary.remaining_capacity.should eq(2)
+
+    expect_raises(ArgumentError, "Insufficient capacity: 2 cannot hold 2 elements with an offset of 1") do
```

Comment:
This error seems very unexpected. When I have an array of size `2` I would assume it can be resized to capacity `2`. IMO `ary.resize(ary.size)` should always be a valid operation to shrink the capacity to match the current occupancy.

In fact, we could even consider adding an argless overload (or a default value) for exactly this use case. I think it might be a common story: After doing some stuff that dynamically builds the array while adding and maybe removing items from it, at some point all processing is done and the array is not going to change. Then it might make sense to resize capacity to the current size. If there have been a lot of removals during the creation process, that could free a lot of memory.

Reply:
My initial code actually had a default value for this, but I decided against it for it being costly: you need to first move the elements to the beginning of the array, and then perform the resize. I thought of `resize` as a lower-level primitive, only for the expert, who can then decide how to act in such cases. But I guess it makes sense to have it nevertheless.